fdb-record-layer-core/src/main/proto/record_metadata.proto

```diff
@@ -95,6 +95,27 @@ message DataStoreInfo {
   // When considering using this, one should also consider introducing a new record type to store the data.
   // This was introduced with FormatVersion.HEADER_USER_FIELDS.
   repeated UserFieldEntry user_field = 8;
+
+  // The state of the RecordCount as defined by record_count_key. This is similar to IndexState, but since there is no
+  // mechanism for building the RecordCount across multiple transactions, the state transitions are slightly different.
+  enum RecordCountState {
+    // The RecordCountKey is maintained and usable for FDBRecordStore.getSnapshotRecordCount
+    READABLE = 1;
```

Comment:
I'm not sure if we care, but if we wanted to be proto3 compatible, we should make this value `READABLE = 0`. That's still not entirely compatible with all proto3 guidance, as they suggest that value 0 be a special "uninitialized" state (which I think would mean that to see if the record count key is readable, we'd need to always check if it equals "uninitialized" or equals a different "readable" value). That being said, we already have a number of proto3 incompatible enums in the `record_metadata.proto` file. I suspect that we'll not end up updating the syntax here, but maybe instead updating it to one of the protobuf "editions", and they already have compatibility mode logic to allow proto2 enums to be integrated into newer code, so we could say that another one of those isn't so bad.

Reply:
I have no strong objections to changing this to `READABLE = 0`, and going up from there. I mostly just copied `Index.Type` below.

Reply:
I took a stab at adding "unspecified" to the enums, and erroring if it is unrecognized, that is in a different branch, at least for now, but the commit is here: https://github.com/ScottDugas/fdb-record-layer/commit/5dcb5760f62c1697aaec55dc475a69bfa8395f35 more testing is needed.

Reply:
One thing to note is that I believe the code in this PR currently takes advantage of the fact that calling `getRecordCountState()` on the enum field returns its default value in order to collapse the `READABLE` and unset states. Adding a new value 0 will change the default to `UNSPECIFIED`. That's technically against the protobuf evolution guidelines (if we did this later), but I think it's fine as long as the code is modified to either more explicitly treat `UNSPECIFIED` as a synonym of `READABLE` or if we add additional `hasRecordCountKey` checks if we add this later